portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,105 @@
 
 package com.google.sps.servlets;
 
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
+import java.io.*; 
+import java.util.*; 
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
-  }
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery results = datastore.prepare(query);
+
+        ArrayList<String> users = new ArrayList<>();
+        ArrayList<String> comments = new ArrayList<>();
+        ArrayList<String> datetimes = new ArrayList<>();
+
+        for (Entity entity : results.asIterable()) {
+            String user = (String) entity.getProperty("user");
+            String comment = (String) entity.getProperty("comment");
+            String datetime = (String) entity.getProperty("datetime");
+
+            users.add(user);
+            comments.add(comment);
+            datetimes.add(datetime);
+        }
+
+        String json = convertToJson(users, comments, datetimes);
+        
+        response.setContentType("application/json;");
+        response.getWriter().println(json);
+    }
+
+    private String convertToJson(ArrayList<String> users, ArrayList<String> comments, ArrayList<String> datetimes) {
+        String json = "{";
```

Comment:
In Java, String is immutable, so concatenating strings to it using + will create new objects, which is not efficient. In such cases it is suggested to use StringBuilder using which we can create mutable object, append into it and finally build it as string. You can refer to StringBuilder examples to know how to use it. FOr example - https://www.geeksforgeeks.org/stringbuilder-class-in-java-with-examples/